packages/frontend-plugin-api/src/wiring/createExtension.ts

```diff
@@ -397,9 +397,7 @@ export function createExtension<
       ),
     );
 
-  return {
-    $$type: '@backstage/ExtensionDefinition',
-    version: 'v2',
+  return OpaqueExtensionDefinition.createInstance('v2', {
     T: undefined as unknown as T,
```

Comment:
Could this be possible to make skippable, as in, being returned by `createInstance` instead as part of the "standard" of these types? It's a bit of an ugly quirk to have to write it this way. Maybe `createInstance<T>` if it's still needed?

Reply:
For now the `T` here is a bit of an outlier, there aren't many other opaque types that use this exact pattern so I think it's best to hold off on it for now. It's also not the type itself but rather the parameters, which we can't really extract from within `createInstance`. I did actually have `createInstance<...>(...)` to start but the version selection was a bit wonky with `undefined` versions so I switched over to this explicit version selection instead, which ofc prevents us from mixing inferred and provided type arguments.